### PR TITLE
feat(stats): add support for canonical service labels

### PR DESCRIPTION
**Pull request by douglas-reid**
**What this PR does / why we need it**:

This PR adds support to the `stats` extension for reporting `{source|destination}_canonical_service` labels to Istio metrics.

This is part of the approved proposal for Istio Canonical Service.  Note: the actual stats config in istio/istio bootstrap will need to be updated to match.

Signed-off-by: Douglas Reid <douglas-reid@users.noreply.github.com>

### CODE REVIEW

File: test/envoye2e/http_metadata_exchange/testoutput/client.yaml [hunk 1]
```diff
@@ -102,7 +106,7 @@ static_resources:
             address:
               socket_address:
                 address: 127.0.0.1
-                port_value: 20083
+                port_value: 20103
```
Comment (kyessenov) [inline test/envoye2e/http_metadata_exchange/testoutput/client.yaml]: What is this change for?

Comment (douglas-reid) [reply test/envoye2e/http_metadata_exchange/testoutput/client.yaml]: ??? generated file. Got changed by running that test.

**Pull request merged**